lib/unique_for_owner_validator.rb

```diff
@@ -13,8 +13,8 @@ def validate(record)
     end
   end
 
-  def unique_fields
-    %i(display_name)
+  def unique_fields_for(record)
```

Comment:
I'm indifferent really.

I feel like this is a bit of a hack anyway.  The custom validator probably ought to be replaced with a scoped uniqueness validation.